public/react/Configure/components/Teams/services/team-configuration/team-configuration/fetcher.ts

```diff
@@ -29,5 +29,34 @@ export const getTeamData = (teamName: string, allTeamData: AllTeamData): TeamCon
 };
 
 export const updateAllTeamData = async (storage: AppStorage, updates: AllTeamData): Promise<void> => {
-  return storage.update(allTeamDataKey, updates);
+  const sanitized = entriesFlatMap(
+    updates,
+    ([teamConfigKey, updatedTeamConfig]: [string, TeamConfiguration | undefined]) => {
+      if (!updatedTeamConfig) return [];
+
+      const value = entriesFlatMap(
+        updatedTeamConfig,
+        ([configKey, updatedConfig]: [string, Configuration | undefined]) => {
+          if (!updatedConfig) return [];
+
+          const configNoNulls = filterNullValues(updatedConfig);
+          
+          return notEmpty(configNoNulls, [configKey, configNoNulls]);
+        }
+      );
+      return notEmpty(value, [teamConfigKey, value]);
+    }
+  );
+  return storage.update(allTeamDataKey, sanitized);
 };
+
+const entriesFlatMap = <TItem, TMap extends Record<string, TItem>>(
+  updates: TMap,
+  fn: ([key, update]: [string, TItem]) => (string | { [k: string]: any })[][]
+) => Object.fromEntries(Object.entries(updates).flatMap(fn));
+
+const filterNullValues = (obj: Record<string, any>) => {
+  return Object.fromEntries(Object.entries(obj).filter(([_, value]) => !!value));
+};
+
+const notEmpty = <TObj extends {}, TResult>(obj: TObj, result: TResult) => (Object.keys(obj).length ? [result] : []);
```

Comment:
added some helpers here to keep the updateAllTeamData a bit cleaner

Reply:
same generic comments as above